### PR TITLE
Fix a class casting error in concurrent writer when enabling AQE [databricks]

### LINKED ISSUE #6762: [BUG] The concurrent writer throws a class casting error when enabling AQE.
**Describe the bug**
@HaoYang670 found this bug, thanks.
The concurrent writer throws a class casting error when enabling AQE.

Error is:
```
Caused by: java.lang.ClassCastException: org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec cannot be cast to com.nvidia.spark.rapids.GpuExec
```

Related [code](https://github.com/NVIDIA/spark-rapids/blob/branch-22.12/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatWriter.scala#L226) is:

```
        if (concurrentWritersEnabled) {
          (empty2NullPlan.executeColumnar(),
              Some(GpuConcurrentOutputWriterSpec(maxWriters, empty2NullPlan.asInstanceOf[GpuExec],
                orderingExpr)))
        } else {
```

If AQE is enabled, `empty2NullPlan` is actually a `AdaptiveSparkPlanExec`.

**Steps/Code to reproduce bug**
```
@ignore_order
@allow_non_gpu(any = True)
def test_my_test(spark_tmp_path):
    data_path = spark_tmp_path + '/PARQUET_DATA'
    assert_gpu_and_cpu_writes_are_equal_collect(
        lambda spark, path: get_25_partitions_df(spark)  # df has 25 partitions for (c1, c2)
            .repartition(2)
            .write.mode("overwrite").partitionBy('c1', 'c2').parquet(path),
        lambda spark, path: spark.read.parquet(path),
        data_path,
        copy_and_update(
            # 10 < 25, will fall back to single writer
            {"spark.sql.maxConcurrentOutputFileWriters": 10},
            {"spark.sql.adaptive.enabled": "true"},    #  enable AQE
        ))
```

### PR DESCRIPTION
**Pull request by res-life**
Closes #6762 

See the description of this issue, the bug is caused by `asInstanceOf` casting.
In this PR:
- Remove this `asInstanceOf` casting. 
- Add a test for enabling AQE.

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
build

**Review by jlowe — DISMISSED** [review]

File: integration_tests/src/main/python/parquet_write_test.py [hunk 1]
```diff
@@ -523,7 +523,9 @@ def test_concurrent_writer(spark_tmp_path):
 
 @ignore_order
 @pytest.mark.skipif(is_before_spark_320(), reason="is only supported in Spark 320+")
-def test_fallback_to_single_writer_from_concurrent_writer(spark_tmp_path):
+@allow_non_gpu(any=True)
+@pytest.mark.parametrize('enable_aqe', ["true", "false"])
```
Comment (gerashegalov) [inline integration_tests/src/main/python/parquet_write_test.py:527]: not a blocker: I'd prefer type-safe Boolean literals `True`, `False`

This is at least the fourth naming for this kind of pytest parameter after `aqe_enabled`, `aqeEnabled` and `adaptive` I see in our code base. Let us standardize on the more common `aqe_enabled` for easier grepping

Comment (res-life) [reply integration_tests/src/main/python/parquet_write_test.py]: Done.

**Comment by res-life** [comment]
build

**Review by gerashegalov — APPROVED** [review]
LGTM

**Comment by res-life** [comment]
Thanks Gera

**Review by gerashegalov — APPROVED** [review]

**Pull request merged**